ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java

```diff
@@ -30,8 +31,27 @@
 
     private static VungleManager sInstance;
 
-    private ConcurrentHashMap<String, VungleBanner> activeBannerAds;
-    private ConcurrentHashMap<String, VungleNativeAd> activeNativeAds;
+    static class BannerRequest {
+        private String placementId;
+        private String uniquePubRequestId;
+
+        BannerRequest(String placementId, String uniquePubRequestId) {
+            this.placementId = placementId;
+            this.uniquePubRequestId = uniquePubRequestId;
+        }
+
+        String getPlacementId() {
+            return placementId;
+        }
+
+        String getUniquePubRequestId() {
+            return uniquePubRequestId;
+        }
+    }
+
+    private ConcurrentHashMap<BannerRequest, VungleBanner> activeBannerAds;
+    private ConcurrentHashMap<BannerRequest, VungleNativeAd> activeNativeAds;
+    private Set<String> mProgressPlacements;
```

Comment:
Hi @HuFangshuai, why can't we use activeBannerAds/activeNativeAds for tracking "progress placements" or why the separate set is needed?

Reply:
Want to fix this scenario: multiple banner requests at the same time for the first time if there's no active banner instance. Try to make sure only the first banner can load and display, and throw away other requests for this placement for the first time.
If without this mProgressPlacements, all these requests will go to load and display progress for the first time. Maybe the second banner could display.
When any banner displayed, then we can use activeBannerAds/activeNativeAds for tracking.
I am not sure whether we should handle this scenario.
@alexander-gubov @palrahul

Reply:
new designed VungleBannerAdapter.